### PR TITLE
MOM_interface changes for regional setups

### PR DESCRIPTION
**Pull request by manishvenu**
Details:
Changes to the configuration for regional setups. This includes new PE layouts, new compsets, and a new xml variable to turn on regional defaults, which currently only changes MOM_input. The bulk of the new changes are to MOM_input.

### CODE REVIEW

File: param_templates/MOM_input.yaml [hunk 1]
```diff
@@ -3605,11 +3676,17 @@ Global:
             else: "PPM_CW"
     MLE_USE_PBL_MLD:
         description: |
-            "TODO"
+            "TODO
```
Comment (alperaltuntas) [inline param_templates/MOM_input.yaml]: Remove TODO?

**Comment by alperaltuntas** [comment]
@manishvenu  Is this PR ready for testing? Any reason to keep it draft, or is it ready to be reviewed?

File: param_templates/MOM_input.yaml (hunk 1 above)
Comment (manishvenu) [reply param_templates/MOM_input.yaml]: Hey Alper, I wasn't sure if this was just for the description, if it is, happy to remove it

Comment (alperaltuntas) [reply param_templates/MOM_input.yaml]: yes, i think it's just for the description.

File: cime_config/config_component.xml [hunk 2]
```diff
@@ -204,6 +205,19 @@
     </desc>
   </entry>
 
+    <entry id="MOM6_DOMAIN_TYPE">
+    <type>char</type>
+    <valid_values>REGIONAL,GLOBAL</valid_values>
+    <default_value>GLOBAL</default_value>
+    <values>
+      <value compset="_MOM6%REGIONAL">REGIONAL</value>
```
Comment (mnlevy1981) [inline cime_config/config_component.xml]: If you change this to

```diff
-<value compset="_MOM6%REGIONAL">REGIONAL</value>
+<value compset="_MOM6%[^_]*REGIONAL">REGIONAL</value>
```

Then you are not required to put `%REGIONAL` as the first compset modifier (`MOM6%REGIONAL%MARBL-BIO` would be equivalent to `MOM6%MARBL-BIO%REGIONAL`)

File: cime_config/config_pes.xml [hunk 3]
```diff
@@ -39,6 +39,140 @@
         </rootpe>
       </pes>
     </mach>
+    <mach name="derecho">
+      <pes pesize="any" compset="_DATM.+_SICE.*_MOM6%REGIONAL(?!.*%MARBL-BIO).+_SWAV">
```
Comment (mnlevy1981) [inline cime_config/config_pes.xml]: Same thing, I think you want `MOM6%[^_]*REGIONAL` here

File: cime_config/config_pes.xml [hunk 4]
```diff
@@ -39,6 +39,140 @@
...
+          <rootpe_rof>0</rootpe_rof>
+          <rootpe_cpl>0</rootpe_cpl>
+          <rootpe_ice>0</rootpe_ice>
+          <rootpe_ocn>128</rootpe_ocn>
+          <rootpe_lnd>0</rootpe_lnd>
+          <rootpe_wav>0</rootpe_wav>
+          <rootpe_glc>0</rootpe_glc>
+        </rootpe>
+      </pes>
+      <pes pesize="any" compset="_DATM.+_SICE.*_MOM6%REGIONAL[^_]*MARBL-BIO.+_SWAV">
```
Comment (mnlevy1981) [inline cime_config/config_pes.xml]: this one is trickier, because you need an or statement to capture when `MARBL-BIO` precedes `REGIONAL`, but I would start with `MOM6%[^_]*REGIONAL[^_]*MARBL-BIO`. The right answer might be `MOM6%([^_]*REGIONAL[^_]*MARBL-BIO|[^_]*MARBL-BIO[^_]*REGIONAL)` but I'm not super confident in that

File: cime_config/config_pes.xml (hunk 3 above)
Comment (mnlevy1981) [reply cime_config/config_pes.xml]: hmm, you might need additional logic to rule out `MOM6%MARBL-BIO%REGIONAL`. What a mess

File: cime_config/config_pes.xml [hunk 5]
```diff
@@ -39,6 +39,140 @@
...
+          <rootpe_rof>0</rootpe_rof>
+          <rootpe_cpl>0</rootpe_cpl>
+          <rootpe_ice>0</rootpe_ice>
+          <rootpe_ocn>128</rootpe_ocn>
+          <rootpe_lnd>0</rootpe_lnd>
+          <rootpe_wav>0</rootpe_wav>
+          <rootpe_glc>0</rootpe_glc>
+        </rootpe>
+      </pes>
+      <pes pesize="any" compset="_DATM.+_CICE.*_MOM6%REGIONAL(?!.*%MARBL-BIO).+_SWAV">
```
Comment (mnlevy1981) [inline cime_config/config_pes.xml]: same as one of the above comments

File: cime_config/config_pes.xml [hunk 6]
```diff
@@ -39,6 +39,140 @@
...
+          <rootpe_rof>0</rootpe_rof>
+          <rootpe_cpl>0</rootpe_cpl>
+          <rootpe_ice>128</rootpe_ice>
+          <rootpe_ocn>256</rootpe_ocn>
+          <rootpe_lnd>0</rootpe_lnd>
+          <rootpe_wav>0</rootpe_wav>
+          <rootpe_glc>0</rootpe_glc>
+        </rootpe>
+      </pes>
+      <pes pesize="any" compset="_DATM.+_CICE.*_MOM6%REGIONAL[^_]*MARBL-BIO.+_SWAV">
```
Comment (mnlevy1981) [inline cime_config/config_pes.xml]: same as one of the above comments

**Review by mnlevy1981 — CHANGES_REQUESTED** [review]
If the regular expressions in `config_pes.xml` get to be too much, we can punt on that... but I do think we want to update `config_component.xml`

File: param_templates/MOM_input.yaml (hunk 1 above)
Comment (manishvenu) [reply param_templates/MOM_input.yaml]: Done

File: cime_config/config_component.xml (hunk 2 above)
Comment (manishvenu) [reply cime_config/config_component.xml]: Done

**Comment by manishvenu** [comment]
Hey @mnlevy,

I updated the regex for the PEs and Claude wrote a quick test script that means it potentially works: /glade/u/home/manishrv/documents/croc/dev/random/test_regex_script.py/glade/u/home/manishrv/documents/croc/dev/random/test_regex_script.py.

It's super confusing! Happy to back out the regex changes (last two commits)

File: cime_config/config_pes.xml (hunk 6 above)
Comment (manishvenu) [reply cime_config/config_pes.xml]: Done

File: cime_config/config_pes.xml (hunk 5 above)
Comment (manishvenu) [reply cime_config/config_pes.xml]: Done

File: cime_config/config_pes.xml (hunk 4 above)
Comment (manishvenu) [reply cime_config/config_pes.xml]: Done - the OR worked.

File: cime_config/config_pes.xml (hunk 3 above)
Comment (manishvenu) [reply cime_config/config_pes.xml]: This worked if you added the exclusion before the Regional.

File: cime_config/config_compsets.xml [hunk 7]
```diff
@@ -120,4 +142,26 @@
...
+
+  <compset>
+    <alias>GR_JRA_GLOFAS</alias>
+    <lname>1850_DATM%JRA_SLND_CICE_MOM6%REGIONAL_DROF%GLOFAS_SGLC_SWAV</lname>
+  </compset>
+
+  <compset>
+    <alias>GR1850MARBL_JRA_GLOFAS</alias>
+    <lname>1850_DATM%JRA_SLND_CICE_MOM6%REGIONAL%MARBL-BIO_DROF%GLOFAS_SGLC_SWAV</lname>
+  </compset>
```
Comment (mnlevy1981) [inline cime_config/config_compsets.xml:165]: @dabail10 and I were talking, and we probably want a CICE modifier as well (e.g. `CICE%REGIONAL`). Now that open boundary conditions are available in CICE, he's going to test that code out but eventually CICE will want to be able to configure regional domains.

Two notes:

1. I assume changing `CICE` to `CICE%<something_else>` will not change answers if CICE's `config_component.xml` doesn't know what to do with the new modifier, but it's worth testing that hypothesis
2. `CICE%REGIONAL` is my first thought, but @dabail10 may propose a better modifier

Comment (manishvenu) [reply cime_config/config_compsets.xml:165]: Maybe this should be another PR? Or can CICE use the MOM6 regional? Because you can't have one without the other?
